src/Symfony/Component/HttpClient/Test/HarFileResponseFactory.php

```diff
@@ -0,0 +1,101 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpClient\Test;
+
+use Symfony\Component\HttpClient\Exception\TransportException;
+use Symfony\Component\HttpClient\Response\MockResponse;
+use Symfony\Contracts\HttpClient\ResponseInterface;
+
+/**
+ * See: https://w3c.github.io/web-performance/specs/HAR/Overview.html.
+ *
+ * @author Gary PEGEOT <garypegeot@gmail.com>
+ */
+class HarFileResponseFactory
+{
+    public function __construct(private string $archiveFile)
+    {
+    }
+
+    public function setArchiveFile(string $archiveFile): void
+    {
+        $this->archiveFile = $archiveFile;
+    }
+
+    public function __invoke(string $method, string $url, array $options): ResponseInterface
+    {
+        if (!is_file($this->archiveFile)) {
+            throw new \InvalidArgumentException(sprintf('Invalid file path provided: "%s".', $this->archiveFile));
+        }
+
+        $json = json_decode(
+            json: file_get_contents($this->archiveFile),
+            associative: true,
+            flags: \JSON_THROW_ON_ERROR
+        );
+
+        foreach ($json['log']['entries'] as $entry) {
```

Comment:
We always just look at the first entry, right? 

Or should we return a generator and yield Responses?

Reply:
In the same order they came in the file?

I though about it, but you most likely want to map a request to a specific entry of that file (one with same method / uri / request body)

Reply:
You are probably correct. The user will probably match uri with request body in most cases. But can we still make it return a Generator? Ie, if a user does include multiple request/responses in the HAR file, we wont silently ignore it. 


-----

Btw, since the HAR file includes method and request URL... would it be possible to match the request with the response.. or is that a separate PR?

Reply:
the matching is precisely what this loop implements (see the checks below to continue to the next entry.

And the current logic supports multiple requests. When passing a callable (which is the case of this class), it gets invoked for each request to provide a response.

Reply:
That is awesome. 

Consider the following code: 

```php
$factory = new HarFileResponseFactory($myHarFile);
$client = new MockHttpClient($factory);

$client->request(/** A */);
$client->request(/** B */);
```

And then our MockHttpClient: 

https://github.com/symfony/symfony/blob/d5423dbabd68370190dec217cd7b833027308a0c/src/Symfony/Component/HttpClient/MockHttpClient.php#L62-L85

I thought the callable `$this->responseFactory` was reset after each call to `MockHttpClient::request()`. Since it isn't, it makes no sense to return a generator. 

Sorry for being slow and thank you for the explanations. Im super happy with this PR.